src/components/DetailSection.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { RootState } from '../redux/reducers'
+import LoadingSpinner from './LoadingSpinner'
+import DisplayList from './DisplayList'
+import FailAlert from './FailAlert'
+
+type DisplaySectionProps = RootState['details'];
```

Comment:
this looks confusing, can we have a type defined here itself?

Reply:
Added automatically inferenced types.